src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala

```diff
@@ -72,12 +71,14 @@ case class BlockFetcherState(
 
   def takeHashes(amount: Int): Seq[ByteString] = waitingHeaders.take(amount).map(_.hash)
 
-  def appendHeaders(headers: Seq[BlockHeader]): BlockFetcherState =
-    withPossibleNewTopAt(headers.lastOption.map(_.number))
-      .copy(
-        waitingHeaders = waitingHeaders ++ headers.filter(_.number > lastBlock).sortBy(_.number),
-        lastBlock = HeadersSeq.lastNumber(headers).getOrElse(lastBlock)
-      )
+  def appendHeaders(headers: Seq[BlockHeader]): Either[String, BlockFetcherState] =
+    validatedHeaders(headers).map(validHeaders => {
+      withPossibleNewTopAt(headers.lastOption.map(_.number))
+        .copy(
+          waitingHeaders = waitingHeaders ++ headers.filter(_.number > lastBlock).sortBy(_.number),
```

Comment:
Since this filter was a thing, which caused inconsistencies I removed it. So far tests are passing, I'm going to run cluster and sync to Mordor to see whether it affects stability in any way. I can image one case where it does, but I'm not sure whether it's really an issue

Reply:
I never liked that filter too much :thinking: that filtering will be done by the importer either way, or maybe could be moved to when the fetcher is queried (to decouple them further)